### ISSUE TITLE
New feature - ability to save/restore settings

### ISSUE THREAD

**Issue by springfall2008** [springfall2008/batpred#612]
I'm experimenting with a feature that creates a file called '/homeassistant/predbat_restore_settings.yaml' which contains the current user configuration.

If you copy the contents of this into a new automation in HA it will, when run, restore your current HA settings for Predbat.

The idea is then users can save one or more sets of configuration and quickly change between them.

The alternative is to use a select drop down and have the user be able to save a new setting or restore an old one based on slots or suchlike.

Any feedback?

**Comment by springfall2008** [comment]
I should have said the above is live in the current release but needs testing and documentation

**Comment by PianSom** [comment]
Thanks for all the work - I am still getting to terms with the many, many options and trying to work out which ones work best for me. So a quick way to save/change settings would be fantastic.

I am a Home Assistant Core user, with Appdaemon in a different Docker container. Just as a note, the settings .yaml  is found in the appdeamon root folder (the one above "apps"). (I found the dashboard yaml there too - was wondering where that was!)

Personally, I'd have a preference for the drop-down alternative - saves me having to do the cli shuffle. But I'd be happy to manage the .yaml's with a cli and then have a drop-down populated from a dedicated profile folder if that was easier.

**Comment by dochtie** [comment]
Thanks for this! Great idea. Could you also add a dropdown for “DEFAULT”? An easy way to quickly restore all settings back to the defaults.

**Comment by Boffinboy** [comment]
I agree this would be quite useful, including the default option. Drop down would be preferable. Thank you for the continued improvements!

**Comment by springfall2008** [comment]
I've changed this feature now (it's on main) to use a dropdown menu with save and restore options, please have a look and feedback.

**Comment by PianSom** [comment]
Looks good so far. 

For docs - the backups are stored in a folder called "predbat_save" alongside the "apps" folder.

Is this change in 7.14.44 (and all points forward)?

**Comment by mpartington** [comment]
Seems to work ok for me.  Maybe worth documenting that you can manually save the file name to something more meaningful using file editor (unless there's an easier method I've missed).

Is there a way to switch between the settings via a HA automation, or is it internal to Predbat?  My use case is currently I need a different set of settings for say a DFS session (e.g. turn off combine charge windows), so at the moment my external automation has a number of call service lines to update the relevant fields, then change them back. It would be simpler just to use the profiles. Not critical for me, as it's already set up and working, but maybe of benefit as a feature.

**Comment by PianSom** [comment]
I am just back from holiday, so have `holidays_days_left` set. I see this is in the saved settings file.  I wonder if this is a setting that one might actually in practise want to save + restore?

**Comment by springfall2008** [comment]
@mpartington by pop-ups do you mean notifications?

@PianSom actually its a good point, I probably shouldn't restore holiday_days_left , or the iboost status or the manual car soc level

**Comment by mpartington** [comment]
Yes, notifications

**Comment by mpartington** [comment]
Seeing a bug I suspect is related to this. Everytime I restart HA, it seems to change my settings to a different saved configuration

**Comment by springfall2008** [comment]
Do you have a log, it should be recorded?

**Comment by mpartington** [comment]
Actually it doesn't appear to be all settings, but read only and min metric improvement definitely changed following restart at 18:46.  Attached log, but can turn debug on if required

[appdaemon1.log](https://github.com/springfall2008/batpred/files/13998935/appdaemon1.log)
[access.log](https://github.com/springfall2008/batpred/files/13998936/access.log)

**Comment by mpartington** [comment]
As usual, this was nothing to do with Predbat...

Appears what is happening is the octopus savings session binary sensor is briefly switched on following a reboot. When it switches off, it triggers a local automation to put Predbat into a particular combination of settings.

**Comment by gcoan** [comment]
Yes I found the same had happened, my Christmas lights were on at 1:30 am. Looked at the automation traces and found it was the binary sensor for saving sessions that had fired

**Comment by gcoan** [comment]
@mpartington I have raised this as a bug on the Octopus integration https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/716

**Comment by mpartington** [comment]
Brilliant, thanks! As a workaround, I've added a helper that I switch on myself upon detection/signup of a saver session and turn it off at the end of a session as the last thing.  I've added a condition that that sensor must be true to restore a predbat config. Has the bonus I can now call up a config for the session (that will always be the same) and just revert to the ' previous.yaml' that has my latest preferences.
